File: ilina_ka/app/src/main/java/com/example/ilina_ka/CardListActivity.java
package com.example.ilina_ka;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class CardListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_card_list2);

        final TextView tvTitle = findViewById(R.id.tvTitleLD);
        final TextView tvId = findViewById(R.id.tvIdLD);
        final TextView tvPhoto = findViewById(R.id.tvPotLD);

        Bundle argument = getIntent().getExtras();
        if (argument == null){
           return;
        }

        Card card = (Card) argument.getParcelable(Card.class.getSimpleName());
        if (card == null){
           return;
        }

        tvTitle.setText(card.getTitle());
        tvId.setText(card.getId());
        tvPhoto.setText(card.getPhoto());
    }
}

File: app/src/main/java/com/example/homework_1/CardListActivity.kt
package com.example.homework_1

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.content.ContextCompat.startActivity
import android.view.View
import kotlinx.android.synthetic.main.activity_card_list.*


class CardListActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_card_list)
        textView3.visibility = View.INVISIBLE
        textView4.visibility = View.INVISIBLE
        textView5.visibility = View.INVISIBLE
        imageView.visibility = View.INVISIBLE





    }



    fun AddNewCard_Click(view: View) {

        val intent1 = Intent(this, EditCardActivty::class.java)
        startActivity(intent1)
    }




    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

       // if (resultCode == Activity.RESULT_OK) {
            var arguments = intent.extras
            var card: Card

                card = arguments.getParcelable(Card::class.java.simpleName) as Card

                textView3.text = card.name
                textView4.text = card.category
                textView5.text = "Скидка " + card.percent.toString() + "%"
                textView.visibility = View.INVISIBLE
                textView2.visibility = View.INVISIBLE
                textView3.visibility = View.VISIBLE
                textView4.visibility = View.VISIBLE
                textView5.visibility = View.VISIBLE
                imageView.visibility = View.VISIBLE

       // }


    }



}
